### PR TITLE
Pagination sorting swagger

### PR DESCRIPTION
**Pull request by VolandevlodD**
Added Pagination, Sorting, and Swagger to the existing controllers

### CODE REVIEW

**Review by rel1c-hub — APPROVED** [review]
LGTM!
Well done

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -47,6 +48,8 @@ public BookDto update(Long id, CreateBookRequestDto requestDto) {
     @Override
     public void deleteById(Long id) {
         checkIfBookExists(id);
+        bookRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
```
Comment (oksana-miazina) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java]: You check entity for existing twice here

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -47,6 +48,8 @@ public BookDto update(Long id, CreateBookRequestDto requestDto) {
     @Override
     public void deleteById(Long id) {
         checkIfBookExists(id);
```
Comment (oksana-miazina) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java]: You can also add the same checking for ISBN in create method (what if I try to create a new book with existing ISBN)

**Review by oksana-miazina — APPROVED** [review]
Awesome!

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**